### PR TITLE
Delaying map first paint until we are done with animations.

### LINKED ISSUE #44: Postpone first render of MagnetMap until any transitions have completed
The initial render of `MagnetMap` is quite resource intensive. If it is done when animations/transitions are in progress we can drop frames.

[The react-native docs](https://facebook.github.io/react-native/docs/performance.html#slow-navigator-transitions) shows how this can be done.

### PR DESCRIPTION
**Pull request by arcturus**
Fixes #44

### CODE REVIEW

**Comment by arcturus** [comment]
@wilsonpage f?

I know that the current placeholder is nothing, but I cannot appreciate it on my devices, and want to check if it ease the pain in yours.

File: js/components/MagnetMap/index.js [hunk 1]
```diff
@@ -6,23 +6,58 @@ import MagnetMapMarker from './MagnetMapMarker';
...
     return (
       <View style={[styles.root, style]}>
-        <MapView
-          provider={MapView.PROVIDER_GOOGLE}
-          style={styles.map}
-          region={region}
-          customMapStyle={mapStyle}>
-          {children}
-        </MapView>
+        {map}
```
Comment (wilsonpage) [inline js/components/MagnetMap/index.js]: In other components we're using the pattern:

```js
{this.renderMap()}
```

File: js/components/MagnetMap/index.js [hunk 2]
```diff
@@ -6,23 +6,58 @@ import MagnetMapMarker from './MagnetMapMarker';
...
       </View>
     );
   }
+
+  renderMap() {
+    if (this.state.usePlaceHolder) {
+      return this.renderPlaceHolder();
```
Comment (wilsonpage) [inline js/components/MagnetMap/index.js]: You can just `return` here, unless you have a placeholder to show?

```js
if (this.state.usePlaceHolder) {
  return;
}
```

**Review by wilsonpage — APPROVED** [review]
Looks good! Some small nitty wits.

File: js/components/MagnetMap/index.js (hunk 2 above)
Comment (arcturus) [reply js/components/MagnetMap/index.js]: Right, i did that thinking that maybe in the future, we can have like a real placeholder (image, spinner, whatever).

But can let the simplified version and then expand to what we have now if necessary.

**Pull request merged**